extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/tracing/WithSpanJob.java

```diff
@@ -0,0 +1,53 @@
+package io.quarkus.quartz.runtime.tracing;
+
+import jakarta.enterprise.inject.spi.CDI;
+
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.InstrumenterBuilder;
+import io.opentelemetry.instrumentation.api.instrumenter.code.CodeAttributesExtractor;
+
+public class WithSpanJob implements Job {
+    private final Job delegate;
+
+    private final Instrumenter<JobExecutionContext, Void> instrumenter;
+
+    public WithSpanJob(Job delegate, boolean captureExperimentalSpanAttributes) {
+        this.delegate = delegate;
+        InstrumenterBuilder<JobExecutionContext, Void> jobInstrumenterBuilder = Instrumenter.builder(
+                CDI.current().select(OpenTelemetry.class).get(), "io.quarkus.opentelemetry",
```

Comment:
It might make more sense to add an `OpenTelemetry` parameter to the constructor and provide the instance in the component that instantiates `WithSpanJob`...